sale_procurement_group_by_requested_date/models/sale_order.py

```diff
@@ -9,12 +9,13 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    def _prepare_order_line_procurement(self, cr, uid, order, line,
-                                        group_id=False, context=None):
-        values = super(SaleOrder, self)._prepare_order_line_procurement(
-            cr, uid, order, line, group_id=group_id, context=context)
-        if line.requested_date:
-            req_datetime = fields.Datetime.from_string(line.requested_date)
+    @api.multi
+    def _prepare_order_line_procurement(self, group_id=False):
+        values = super(SaleOrder, self).\
+            _prepare_order_line_procurement(group_id=group_id)
+        self.ensure_one()
+        if self.requested_date:
```

Comment:
The method earlier was as per the older version

Now, self is referring to the sale order lines and the parameters used are as per [newer version](https://github.com/odoo/odoo/blob/9.0/addons/sale/sale.py#L565).